CRM/Contact/BAO/Relationship.php

```diff
@@ -300,6 +300,14 @@ public static function add(&$params, $ids = array(), $contactId = NULL) {
     if ($type == 6) {
       CRM_Contact_BAO_Household::updatePrimaryContact($params['contact_id_b'], $params['contact_id_a']);
     }
+    //Clear current employer if relationship type is updated.
+    if (!empty($relationshipId)) {
+      $existingTypeID = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Relationship', $relationshipId, 'relationship_type_id');
+      $existingTypeName = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', $existingTypeID, 'name_b_a');
+      if ($existingTypeName == 'Employer of' && (empty($params['is_active']) || empty($params['is_current_employer']) || $existingTypeID != $type)) {
```

Comment:
In the above format - it fails on the form layer and doesn't remove employer_id when `is_current_employer ` is unchecked.

The change is now modified to look for 'is_current_employer' variable on the form itself instead of BAO and clear the employer if it is not enabled. Tested various scenarios and looks fine to me.